src/Symfony/Component/Workflow/Dumper/PlantUmlDumper.php

```diff
@@ -0,0 +1,198 @@
+
+    protected static $defaultOptions = array(
+        'skinparams' => array(
+            'titleBorderRoundCorner' => 15,
+            'titleBorderThickness' => 2,
+            'state' => array(
+                'BackgroundColor<<'.self::INITIAL.'>>' => '#87b741',
+                'BackgroundColor<<'.self::MARKED.'>>' => '#3887C6',
+                'BorderColor' => '#3887C6',
+                'BorderColor<<'.self::MARKED.'>>' => 'Black',
+                'FontColor<<'.self::MARKED.'>>' => 'White',
+            ),
+            'agent' => array(
+                'BackgroundColor' => '#ffffff',
+                'BorderColor' => '#3887C6',
+            ),
+        ),
+    );
+
+    protected $transitionType = self::DEFAULT_TRANSITION_TYPE;
+
+    public function __construct($transitionType = null)
+    {
+        if (static::SQUARE_TRANSITION_TYPE === $transitionType) {
+            $this->transitionType = $transitionType;
+        }
+    }
+
+    protected function isSquareTransitionType()
+    {
+        return static::SQUARE_TRANSITION_TYPE === $this->transitionType;
+    }
+
+    /**
+     * @param Definition   $definition
+     * @param Marking|null $marking
+     * @param array        $options
+     */
+    public function dump(Definition $definition, Marking $marking = null, array $options = array())
+    {
+        $options = array_replace_recursive(self::$defaultOptions, $options);
+        $code = $this->initialize($options);
+        foreach ($definition->getPlaces() as $place) {
+            $code[] =
+                "state {$place}".
+                ($definition->getInitialPlace() === $place ? ' <<'.self::INITIAL.'>>' : '').
+                ($marking && $marking->has($place) ? ' <<'.self::MARKED.'>>' : '');
+        }
+        if ($this->isSquareTransitionType()) {
+            foreach ($definition->getTransitions() as $transition) {
+                $code[] = "agent {$transition->getName()}";
+            }
+        }
+        foreach ($definition->getTransitions() as $transition) {
+            foreach ($transition->getFroms() as $from) {
+                foreach ($transition->getTos() as $to) {
+                    if ($this->isSquareTransitionType()) {
+                        $lines = array(
+                            "{$from} --> {$transition->getName()}",
+                            "{$transition->getName()} --> {$to}",
+                        );
+                        foreach ($lines as $line) {
+                            if (!in_array($line, $code)) {
+                                $code[] = $line;
+                            }
+                        }
+                    } else {
+                        $code[] = "{$from} --> {$to}: {$transition->getName()}";
+                    }
+                }
+            }
+        }
+
+        return $this->startPuml($options).$this->getLines($code).$this->endPuml($options);
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return string
```

Comment:
PHP doc be removed (like others).
You can add the return type hint :)

(actually, PHPDoc do not add value. That's why I'm asking to remove it)

Reply:
done